src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -45,6 +47,10 @@ public CommandResult execute(String commandText) {
         final CommandResult result = command.execute();
         updateConfigStorage(oldConfig);
         updateTaskBookStorage(taskBookListener);
+
+        if (model.hasUncommittedChanges() && !(command instanceof UndoCommand) && !(command instanceof RedoCommand) ) {
```

Comment:
ive not tested that out, never considered that reverting back to an earlier commit would have no diff between current TaskBook and head. Yes that seems very logical